src/main/java/org/jenkinsci/deprecatedusage/Downloader.java

```diff
@@ -98,33 +108,53 @@ public void run() {
 
         private void doRun() throws IOException, DigestException {
             URL url = new URL(file.getUrl());
-            HttpURLConnection request = (HttpURLConnection) url.openConnection();
```

Comment:
the old code did not retry in the presence of a Socket issue (Connect failure / timeout etc).